core/src/execution/evm.rs

```diff
@@ -62,20 +60,55 @@ impl<N: NetworkSpec, R: ExecutionRpc<N>> Evm<N, R> {
     }
 
     pub async fn estimate_gas(&mut self, tx: &N::TransactionRequest) -> Result<u64, EvmError> {
-        let tx = self.call_inner(tx).await?;
+        let (result, ..) = self.call_inner(tx, false).await?;
 
-        match tx.result {
+        match result {
             ExecutionResult::Success { gas_used, .. } => Ok(gas_used),
             ExecutionResult::Revert { gas_used, .. } => Ok(gas_used),
             ExecutionResult::Halt { gas_used, .. } => Ok(gas_used),
         }
     }
 
-    async fn call_inner(&mut self, tx: &N::TransactionRequest) -> Result<ResultAndState, EvmError> {
+    pub async fn create_access_list(
+        &mut self,
+        tx: &N::TransactionRequest,
+    ) -> Result<AccessListResultWithAccounts, EvmError> {
+        let (result, accounts) = self.call_inner(tx, false).await?;
```

Comment:
This is used to create the extended access list in the server, so we should be able to designate whether to use `simulate` as true or false, so that we get the right validation here.

Reply:
`/createExtendedAccessList` now also accepts `validateTx: bool` in request body to allow control over this. https://github.com/a16z/helios/pull/505/commits/77b838d790f8caa685e8da6a0ea6d74ab51ce81b.